hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java

```diff
@@ -375,6 +389,45 @@ public InputSplit[] getSplits(JobConf job, int numSplits) throws IOException {
     // clear work from ThreadLocal after splits generated in case of thread is reused in pool.
     Utilities.clearWorkMapForConf(job);
 
+    // build internal schema for the query
+    if (job.getBoolean(INTERNAL_SCHEMA_CACHE_ENABLE, true)) {
```

Comment:
Should it always be true because this is a deterministic optimization?

Reply:
yes, you are right